detect/detect_test.go

```diff
@@ -603,11 +603,32 @@ func TestFromFiles(t *testing.T) {
 			cfgName:          "simple",
 			expectedFindings: []report.Finding{},
 		},
+		{
+			source:  filepath.Join(repoBasePath, "nogit", ".env.prod"),
+			cfgName: "generic",
+			expectedFindings: []report.Finding{
+				{
+					Description: "Generic API Key",
+					StartLine:   4,
+					EndLine:     4,
+					StartColumn: 5,
+					EndColumn:   35,
+					Match:       "PASSWORD=8ae31cacf141669ddfb5da",
+					Secret:      "8ae31cacf141669ddfb5da",
+					Line:        "\nDB_PASSWORD=8ae31cacf141669ddfb5da\nDB_NAME=best_db",
```

Comment:
I _think_ this exposes a deeper issue. The `EndLine` and `EndColumn` are now correct but the `line` information was previously populated with the incorrect coordinates.

\+ @zricethezav Shouldn't `line` be a single line? Or is this expected behaviour when a match spans multiple lines?

Reply:
> + @zricethezav Also shouldn't line be a single line? Or is this expected behaviour when a match spans multiple lines?

Line should be a single line, correct